aot/src/main/java/com/dylibso/chicory/experimental/aot/AotCompiler.java

```diff
@@ -688,6 +710,18 @@ private void compileFunction(
 
         List<AotInstruction> instructions = analyzer.analyze(funcId);
 
+        if (hasTooManyParameters(type)) {
+            // unbox the arguments from long[]
+            for (int i = 0; i < type.params().size(); i++) {
+                var param = type.params().get(i);
+                asm.load(0, OBJECT_TYPE);
+                asm.iconst(i);
+                asm.aload(LONG_TYPE);
+                emitLongToJvm(asm, param);
+                asm.store(ctx.localSlotIndex(i), asmType(param));
+            }
+        }
+
         // initialize local variables to their default values
         int localsCount = type.params().size() + body.localTypes().size();
```

Comment:
Should `type.params().size()` be 1 when we use `hasTooManyParameters` ?

Reply:
I wasn't too sure. but I think your right I updated it and the test still passes.